Fitness/Controllers/FilterViewController.swift

```diff
@@ -746,3 +533,224 @@ class FilterViewController: UIViewController, UICollectionViewDelegateFlowLayout
+            collectionView.selectItem(at: indexPath, animated: true, scrollPosition: .right)
+            cell.isSelected = true
+        }
+        return cell
+    }
+}
+
+//MARK: TableViewDataSource
+extension FilterViewController: UITableViewDataSource {
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        var numberOfRows = 0
+        if tableView == instructorDropdown{
+            if(instructorDropdownData.completed == false){
+                return 0
+            }
+            
+            switch instructorDropdownData.dropStatus{
+            case .up:
+                numberOfRows = 0
+            case .half:
+                numberOfRows = 3
+            case .down:
+                numberOfRows = instructorDropdownData.titles.count
+            default:
+                numberOfRows = 0
+            }
+        }else if tableView == classTypeDropdown{
+            if(classTypeDropdownData.completed == false){
+                return 0
+            }
+            
+            switch classTypeDropdownData.dropStatus{
+            case .up:
+                numberOfRows = 0
+            case .half:
+                numberOfRows = 3
+            case .down:
+                numberOfRows = classTypeDropdownData.titles.count
+            default:
+                numberOfRows = 0
+            }
+        }
+        return numberOfRows
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: DropdownViewCell.identifier, for: indexPath) as! DropdownViewCell
+        
+        if tableView == instructorDropdown{
+            if(indexPath.row < instructorDropdownData.titles.count){
+                cell.titleLabel.text = instructorDropdownData.titles[indexPath.row]
+                cell.id = instructorDropdownData.ids[indexPath.row]
+                
+                if(selectedInstructors.contains(cell.id)){
+                    cell.checkBoxColoring.backgroundColor = .fitnessYellow
+                }
+            }
+        }else if tableView == classTypeDropdown{
+            if(indexPath.row < classTypeDropdownData.titles.count){
+                cell.titleLabel.text = classTypeDropdownData.titles[indexPath.row]
+                cell.id = classTypeDropdownData.ids[indexPath.row]
+                if(selectedClasses.contains(cell.id)){
+                    cell.checkBoxColoring.backgroundColor = .fitnessYellow
+                }
+            }
+        }
+        return cell
+    }
+}
+
+//MARK: TableViewDelegate
+extension FilterViewController: UITableViewDelegate {
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        let cell = tableView.cellForRow(at: indexPath) as! DropdownViewCell
+        var shouldAppend: Bool
+        
+        if(cell.checkBoxColoring.backgroundColor == .fitnessYellow){
```

Comment:
This if else check is weird and can be condensed. You can make `var shouldAppend = cell.checkBoxColoring.backgroundColor != .fitnessYellow` and do the same for the background color